awscli/customizations/s3/s3handler.py

```diff
@@ -367,169 +381,128 @@ def _enqueue_upload_end_task(self, filename, upload_context):
         self.executor.submit(complete_multipart_upload_task)
 
 
-class S3StreamHandler(S3Handler):
+class S3TransferStreamHandler(BaseS3Handler):
     """
     This class is an alternative ``S3Handler`` to be used when the operation
     involves a stream since the logic is different when uploading and
     downloading streams.
     """
-    # This ensures that the number of multipart chunks waiting in the
-    # executor queue and in the threads is limited.
-    MAX_EXECUTOR_QUEUE_SIZE = 2
-    EXECUTOR_NUM_THREADS = 6
+    MAX_IN_MEMORY_CHUNKS = 6
 
     def __init__(self, session, params, result_queue=None,
                  runtime_config=None):
-        if runtime_config is None:
-            # Rather than using the .defaults(), streaming
-            # has different default values so that it does not
-            # consume large amounts of memory.
-            runtime_config = RuntimeConfig().build_config(
-                max_queue_size=self.MAX_EXECUTOR_QUEUE_SIZE,
-                max_concurrent_requests=self.EXECUTOR_NUM_THREADS)
-        super(S3StreamHandler, self).__init__(session, params, result_queue,
-                                              runtime_config)
+        super(S3TransferStreamHandler, self).__init__(
+            session, params, result_queue, runtime_config)
+        self.config = create_transfer_config_from_runtime_config(
+            self._runtime_config)
+
+        # Restrict the maximum chunks to 1 per thread.
+        self.config.max_in_memory_upload_chunks = \
+            self.MAX_IN_MEMORY_CHUNKS
+
+    def call(self, files, manager=None):
```

Comment:
I would prefer to keep the same interface as `BaseS3Handler` and instead have the manager be passed in during the `__init__` if someone wants to override the manager (i.e in your tests).